src/shogun/mathematics/linalg/internal/implementation/Variance.h

```diff
@@ -0,0 +1,319 @@
+	 * \f$\bar k=\frac{1}{m}\sum_{i=1}^m k_i\f$.  This is computed for each column \f$k\f$.
+	 *
+	 * Computes the variance for each column of a matrix
+	 *
+	 * @param m matrix of values
+	 * @return the column-wise variance of a matrix
+	 */
+	static ReturnType compute(SGMatrix<T> m)
+	{
+		REQUIRE(m.num_rows > 0, "Please ensure that m has more than 0 rows.\n")
+		REQUIRE(m.num_cols > 0, "Please ensure that m has more than %d columns.\n")
+
+		
+	}
+};
+
+/**
+ * @Brief A generic class that computes the variance of a matrix row-wise.
+ */
+template<enum Backend, typename Matrix>
+struct rowwise_variance
+{
+	/** Generic scalar type */
+	typedef typename Matrix::Scalar T;
+
+	/** Vector return type */
+	typedef SGVector<T> ReturnType;
+
+	/** Calculates unbiased empirical variance estimate of the given values from each row of
+	 * an input matrix. Given a single row \f$k\f$ with entries \f$\{x_{k1}, ..., x_{kn}\}\f$ 
+	 * from matrix \f$x\f$, this is \f$\frac{1}{n-1}\sum_{i=1}^n (k_i-\bar{k})^2\f$ where
+	 * \f$\bar k=\frac{1}{n}\sum_{i=1}^n k_i\f$.  This is computed for each row \f$n\f$.
+	 *
+	 * Computes the variance for each row of a matrix
+	 *
+	 * @param x matrix of values
+	 * @return the row-wise variance of a matrix
+	 */
+	static ReturnType compute(Matrix x);
+};
+
+/**
+ * @Brief A specialization of colwise_variance that uses SGMatrix and SGVector as its types
+ * and uses Eigen3 as its backend component
+ */
+template<typename Matrix>
+struct rowwise_variance<Backend::EIGEN3, Matrix>
+{
+
+	/** Generic scalar type */
+	typedef typename Matrix::Scalar T;
+
+	/** Vector return type */
+	typedef SGVector<T> ReturnType;
+
+	/** Eigen vector type */
+	typedef Eigen::Matrix<T,Eigen::Dynamic,1> VectorXt;
+
+	/** Eigen matrix type */
+	typedef Eigen::Matrix<T,Eigen::Dynamic, Eigen::Dynamic> MatrixXt;
+
+	/** Calculates unbiased empirical variance estimate of the given values from each row of
+	 * an input matrix. Given a single row \f$k\f$ with entries \f$\{x_{k1}, ..., x_{kn}\}\f$ 
+	 * from matrix \f$x\f$, this is \f$\frac{1}{n-1}\sum_{i=1}^n (k_i-\bar{k})^2\f$ where
+	 * \f$\bar k=\frac{1}{n}\sum_{i=1}^n k_i\f$.  This is computed for each row \f$n\f$.
+	 *
+	 * Computes the variance for each row of a matrix
+	 *
+	 * @param m matrix of values
+	 * @return the column-wise variance of a matrix
+	 */
+	static ReturnType compute(SGMatrix<T> x)
+	{
+		REQUIRE(x.num_rows > 0, "Please ensure that x has more than 0 rows.\n")
+		REQUIRE(x.num_cols > 0, "Please ensure that x has more than 0 columns.\n")
+
+		SGVector<T> tempVec; //used to store multiple results
+		SGMatrix<T> squaredResult(x.num_rows, x.num_cols);
+		Eigen::Map<MatrixXt> eigX = x;
+		Eigen::Map<MatrixXt> eigSquaredResult = squaredResult;
+
+		tempVec = rowwise_mean<Backend::EIGEN3, SGVector<T>>::compute(x, false);
+		Eigen::Map<VectorXt> eigMean = tempVec;
+		for(int i = 0; i < x.num_cols; ++i)
+		{
+			eigSquaredResult.col(i) = eigMean;
```

Comment:
I don't know if there's a better way to do this - I'm trying to make all of the columns of eigMean be equal to tempVec (which holds the rowwise mean for each row)